packages/graph/hash_graph/lib/graph/src/store/crud.rs

```diff
@@ -31,6 +31,23 @@ pub trait Read<T: Send>: Sync {
     /// [`Store`]: crate::store::Store
     async fn read<'f: 'q, 'q>(&self, query: &'f Self::Query<'q>) -> Result<Vec<T>, QueryError>;
 
+    async fn read_one<'f: 'q, 'q>(&self, query: &'f Self::Query<'q>) -> Result<T, QueryError> {
+        let mut records = self.read(query).await?;
+        let record = records.pop().ok_or_else(|| {
+            Report::new(QueryError).attach_printable(
+                "Expected exactly one record to be returned from the query but none was returned",
+            )
+        })?;
+        ensure!(
+            records.is_empty(),
```

Comment:
I remember us discussing this pattern before, is there a reason you're doing this check first instead of checking if the length is more than one first?

Reply:
I mainly did this because the error message differs slightly ("none was returned" vs "<LEN> were returned" with `LEN > 1`). Also, when doing the length-check I need to `pop` anyway and have to somehow unwrap the option.

Reply:
We have a similar pattern elsewhere but there we only required a slice. AFAIK there is no way to destructure a `Vec` into `[element]` (instead `&[element]`).